src/components/Input/BaseInput/ControlledBaseInput.tsx

```diff
@@ -28,22 +28,32 @@ export const ControlledBaseInput: React.FC<ControlledBaseInputProps> = ({
     onFocus,
     focusOnField = false,
     ...props
-}) => {
+}, ref) => {
     const isFilled = value !== '';
-    const inputRef = useRef(null);
+    const inputRef = useRef<HTMLInputElement>(null);
     const classNameProps = clsx('moonstone-baseInput', `moonstone-${size}`, `moonstone-${variant}`, className);
+    let handleClear = onClear || null;
 
     useEffect(() => {
         if (focusOnField && !isDisabled && !isReadOnly) {
             inputRef.current.focus({preventScroll: true});
         }
     }, [focusOnField, isDisabled, isReadOnly]);
 
+    // Merging refs because SearchInput needs inputRef but NumberInput needs ref
```

Comment:
As a future improvement, do you think we can update `SearchInput` to use the same `ref` to remove this part?